Add to macOS deployment a guide to build, package, and distribute app with Codemagic CLI tools (#6085)
flutter/website: src/docs/deployment/macos.md

```

</li>
<li markdown="1">

You'll need to export or create a Mac App Distribution and a Mac Installer
Distribution certificate to perform code signing and package a build archive.

If you have existing [certificates][devportal_certificates], you can export the
private keys by executing the following command for each certificate:

```bash
openssl pkcs12 -in <certificate_name>.p12 -nodes -nocerts | openssl rsa -out cert_key
```

Or you can create a new private key by executing

```bash
ssh-keygen -t rsa -b 2048 -m PEM -f cert_key -q -N ""
```

and later have CLI tools automatically create a new Mac App Distribution and
Mac Installer Distribution certificate. You can use the same private key for
each new certificate.

</li>
<li markdown="1">

Fetch the code signing files from App Store Connect

```bash
app-store-connect fetch-signing-files YOUR.APP.BUNDLE_ID \
    --platform MAC_OS \
    --type MAC_APP_STORE \
    --certificate-key=@file:/path/to/cert_key \
    --create
```

Where `cert_key` is either your exported Mac App Distribution certificate private key
or a new private key which will automatically generate a new certificate. 

</li>
<li markdown="1">

> If you do not have a Mac Installer Distribution certificate, you can create a new certificate by executing  

Reviewer: ```suggestion
If you do not have a Mac Installer Distribution certificate,
you can create a new certificate by executing the following:
```

```bash
app-store-connect create-certificate \
    --type MAC_INSTALLER_DISTRIBUTION \
    --certificate-key=@file:/path/to/cert_key \
    --save
```

> with `cert_key` of the private key you created earlier.

Reviewer: ```suggestion
Use `cert_key` of the private key you created earlier.
```

</li>
<li markdown="1">

Fetch the Mac Installer Distribution certificates:

```bash
app-store-connect list-certificates \
    --type MAC_INSTALLER_DISTRIBUTION \
    --certificate-key=@file:/path/to/cert_key \
    --save
```

</li>
<li markdown="1">

Set up a new temporary keychain to be used for code signing:

```bash
keychain initialize
```

> {{site.alert.warning}}
> **Restore Login Keychain!**
> After running `keychain intialize` you **must** run
> 
> ```bash
> keychain use-login
> ```
> 
> in order set your login keychain as the default to avoid potential authentication issues with apps on your machine.
> {{site.alert.end}}

Reviewer: ```suggestion
{{site.alert.secondary}}
  **Restore Login Keychain!**
  After running `keychain initialize` you **must** run the following:<br>

  `keychain use-login`

  This sets your login keychain as the default to avoid potential
  authentication issues with apps on your machine.
{{site.alert.end}}
```

```bash
xcode-project use-profiles
```

</li>

<li markdown="1">

Install Flutter dependencies:

```bash
flutter packages pub get
```

</li>
<li markdown="1">

Install CocoaPods dependencies:

```bash
find . -name "Podfile" -execdir pod install \;
```

</li>
<li markdown="1">

Enable the Flutter macOS option:

```bash
flutter config --enable-macos-desktop
```

</li>
<li markdown="1">

Build the Flutter macOS project:

```bash
flutter build macos --release
```

</li>
<li markdown="1">

Package the app:

```bash
APP_NAME=$(find $(pwd) -name "*.app")
PACKAGE_NAME=$(basename "$APP_NAME" .app).pkg
xcrun productbuild --component "$APP_NAME" /Applications/ unsigned.pkg

INSTALLER_CERT_NAME=$(keychain list-certificates \
          | jq '.[0]
            | select(.common_name
            | contains("Mac Developer Installer"))
            | .common_name' \
          | xargs)
xcrun productsign --sign "$INSTALLER_CERT_NAME" unsigned.pkg "$PACKAGE_NAME"
rm -f unsigned.pkg 
```

</li>
<li markdown="1">

Publish the packaged app to App Store Connect:

```bash
app-store-connect publish \
    --path "$PACKAGE_NAME"
```

</li>
<li markdown="1">

> As mentioned earlier, don't forget to set your login keychain as the default to avoid authentication issues
> with apps on your machine:

Reviewer: ```suggestion
As mentioned earlier, don't forget to set your login keychain
as the default to avoid authentication issues
with apps on your machine:
```
